Source/WebKit/gtk/tutorial.md

```diff
@@ -0,0 +1,337 @@
+# Create a Web Browser in GTK with WebKit
+
+This is a tutorial for creating a simple web browser with some basic functionalities in GTK with WebKit in C language.
+
+WebKitGTK is a multi-platform toolkit for creating graphical user interfaces; it contains a complete set of widgets to be reused and combined together in development. Among them, it provides a [class@WebKitWebView] widget which can load and display web content, which allows integrating web technologies (HTML, CSS, JavaScript) in your applications&mdash;all the way to implement full-fledged browsers like [GNOME Web](https://wiki.gnome.org/Apps/Web/).
+
+It is recommended to go through GTK's [Getting Started](https://docs.gtk.org/gtk3/getting_started.html) page to learn about the basics of using GTK.
+
+## Coding a simple browser
+
+In our `browse1.c` file, we add the following content to it. In the main function we create a GtkApplication called `app` and connect to it a callback function `on_app_activate`, in which a `web_view` GtkWidget loading a specified URI is created and placed in a window container.
+
+```c
+#include <webkit2/webkit2.h>
+
+static void
+on_app_activate(GtkApplication *app)
+{
+    GtkWidget *web_view = webkit_web_view_new();
+    webkit_web_view_load_uri(WEBKIT_WEB_VIEW(web_view), "https://webkitgtk.org");
+
+    GtkWidget* window = gtk_application_window_new(app);
+    // Adds the web_view widget to the window container.
+    gtk_container_add(GTK_CONTAINER(window), web_view);
+    gtk_window_set_default_size(GTK_WINDOW(window), 800, 600);
+
+    gtk_application_add_window(app, GTK_WINDOW(window));
+    // Recursively shows a widget, and any child widgets (if the widget is a container).
+    gtk_widget_show_all(window);
```

Comment:
This tutorial should be updated to use GTK 4. Besides changes required to make it build, I think the only other change that will affect the documentation is widgets are visible by default now so no longer any need to show it.

Reply:
I think we should merge this, iterate on it later. I just read recently “some documentation is better than no documentation”, and while this is using GTK3, truth is we still support it so the documentation would not be factually wrong or anything like that.

Reply:
I agree. We can clarify the tutorial is for GTK3. Something like `# Create a Web Browser in **GTK3** with WebKit`, or a file name change to tutorial_gtk3.md